x-pack/plugins/spaces/server/lib/request_interceptors/on_post_auth_interceptor.test.ts

```diff
@@ -160,19 +216,35 @@ describe('onPostAuthRequestInterceptor', () => {
         },
       });
 
-      await setupFn(server);
+      // The onRequest interceptor is also included here because the onPostAuth interceptor requires the onRequest
+      // interceptor to parse out the space id and rewrite the request's URL. Rather than duplicating that logic,
+      // we are including the already tested interceptor here in the test chain.
+      initSpacesOnRequestInterceptor({
+        config: (configFn() as unknown) as KibanaConfig,
+        http: httpMock,
+      });
 
-      return await server.inject({
-        method: 'GET',
-        url: path,
-        headers,
+      await root.start();
+
+      const legacyServer = kbnTestServer.getKbnServer(root).server;
+
+      initSpacesOnPostAuthRequestInterceptor({
+        config: (configFn() as unknown) as KibanaConfig,
+        onPostAuth: (handler: any) => legacyServer.ext('onPostAuth', handler),
+        getHiddenUiAppById: (app: string) => null,
+        http: httpMock,
+        log,
+        xpackMain: plugins.xpack_main as XPackMainPlugin,
+        spacesService,
       });
+
+      initKbnServer(http);
+
+      return await kbnTestServer.request.get(root, path);
     };
-  });
+  }, 30000);
```

Comment:
These timeouts are pretty long... do you happen to know what's causing the first test to take so long to run?

Reply:
Yeah they are... I believe it's the call to [`root.start()`](https://github.com/elastic/kibana/blob/454601316227bf23a8cf1964e85cc99402bf30c6/x-pack/plugins/spaces/server/lib/request_interceptors/on_post_auth_interceptor.test.ts#L227) which ends up taking a decent amount of time, because [it's actually starting a server](https://github.com/elastic/kibana/blob/b32f1358b178ad7c292985ac88f922c45a3a652f/src/core/server/root/index.ts#L60). I'm not sure why subsequent runs are faster than the first though